lang/java/reef-runtime-multi/src/main/java/org/apache/reef/runtime/multi/client/MultiRuntimeHDILocalDriverConfigurationProviderImpl.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.runtime.multi.client;
+
+import org.apache.reef.runtime.common.parameters.JVMHeapSlack;
+import org.apache.reef.runtime.hdinsight.client.HDInsightDriverConfiguration;
+import org.apache.reef.runtime.local.client.parameters.MaxNumberOfEvaluators;
+import org.apache.reef.runtime.local.client.parameters.RackNames;
+import org.apache.reef.runtime.local.driver.LocalDriverConfiguration;
+import org.apache.reef.tang.annotations.Parameter;
+import org.apache.reef.tang.formats.ConfigurationModule;
+
+import javax.inject.Inject;
+import java.net.URI;
+import java.util.Set;
+
+/**
+ * MultiRuntime configuration provider for HDI/Local runtime.
+ */
+public final class MultiRuntimeHDILocalDriverConfigurationProviderImpl extends AbstractDriverConfigurationProvider {
```

Comment:
They are public because we need to bind them. For example:
ConfigurationModule cm =  ExtensibleYarnClientConfiguration.getConfigurationModule(params)
            .set(ExtensibleYarnClientConfiguration.DRIVER_CONFIGURATION_PROVIDER,
                    MultiRuntimeYarnLocalDriverConfigurationProviderImpl.class);

In this case I need to bind this implementation to the YarnConfiguration. This way I am changing teh behavior of YarnSubmissionHandler.